File: Python /Transaction.py
from abc import ABC, abstractmethod

class Transaction(ABC): 
    def __init__(self, useraccountnumber, atmscreen, atmbankdatabase):
        self.useraccountnumber = useraccountnumber
        self.atmscreen = atmscreen
        self.atmbankdatabase = atmbankdatabase

    def getaccountnumber(self):
        return self.useraccountnumber 

    def getscreen(self):
        return self.atmscreen

    def getbankdatabase(self):
        return self.atmbankdatabase 

    @abstractmethod 
    def execute(self): 
        pass 

File: Golang/Transaction.go
package main

type Transaction interface {
	execute()
}

type transaction struct {
	accountNumber int
	screen        *Screen
	bankDatabase  *BankDatabase
}

func NewTransaction(userAccountNumber int, atmScreen *Screen, atmBankDatabase *BankDatabase) Transaction {
	return &transaction{
		accountNumber: userAccountNumber,
		screen:        atmScreen,
		bankDatabase:  atmBankDatabase,
	}
}

func (t *transaction) GetAccountNumber() int {
	return t.accountNumber
}

func (t *transaction) GetScreen() *Screen {
	return t.screen
}

func (t *transaction) GetBankDatabase() *BankDatabase {
	return t.bankDatabase
}
